code-review/README.md

```diff
@@ -0,0 +1,119 @@
+Code Review
+===========
+
+This is forked/stolen from ^W^W inspired by
+[Thoughtbot's code review guide](https://raw.githubusercontent.com/thoughtbot/guides/master/code-review/README.md)
+
+A guide for reviewing code and having your code reviewed.
+
+Why code reviews?
+-----------------
+
+Code is reviewed before being merged into `master`. The goal of review is to ensure that changes are delivering business value to our customers - not to perfect our code.
+
+When reviewing, consider whether a change:
+
+* Delivers business value. Understand why it is being made and why it is required.
+* Does what it says it does. Try to break it.
+* Can be maintained by the most junior member of our team.
+* Meets our standards and is well written.
+
+Don't let minor technical issues block shipping.
+
+A reviewer may give +1 on a review. Small fixes need +1, complex fixes and large features +2. A branch should have at least +1 before you hit the green button.
+
+Code Review Mindset
+-------------------
+
+It's important to separate our self-worth from the code we write so that we are always open to feedback that will help us improve.
```

Comment:
Hi Seb. That's a great response: thank you!

On _"a code review should mainly be about its suitability to the problem it is trying to solve and the pros and cons of the chosen implementation over alternatives"_ - I 100% agree completely with this statement.

On _"developing guidelines that we as Mobifiers agree on as best practice will make it clearer what is considered objective or subjective"_ - I also agree with this, with a small _caveat_. The more rules we add, the more complex the process of following those rules becomes. @dbader and I have been using PyLint and ESLint on the web-push code, and it drastically simplifies the process of having code syntactically compliant. I'd summarize it as _if the linter complains, fix it, otherwise **it doesn't matter**_.

Sometimes code reviews frustrate me. For instance, where the process of getting working code rolled out to production is blocked by an apparently endless series of requests for minor tweaks that demonstrably have **no** bearing on whether the code is a reasonable solution to the problem. Or at other times (as we've seen), the effort that should go into attempting to find actual issues is instead diverted into arguments about whether one style or another is more "readable", as though readability was not subjective. Reviews like that, at their worst, damage relations between developers, affect morale and reduce our productivity.

Lastly: _"We should make it clear that reviewing a person's code means criticizing their work..."_. I don't agree with this, because it shouldn't be about **criticizing**. Reviewers help the author find defects in the code. That's not criticizing, it's assisting, providing an alternative point of view, asking for clarification. It's **extremely difficult** to criticize someone's else's work in a way that you can be sure won't have a negative effect. It takes time, care and effort to follow rules (e.g. _phrase your observations as questions, not orders_, _start comments with "I feel"_, and so on). Better to adopt the position that reviewing is helping, not correcting.